agr_java_core/src/main/java/org/alliancegenome/neo4j/repository/Neo4jRepository.java

```diff
@@ -180,5 +180,21 @@ protected String addWhereClauseString(String fieldName, FieldFilter fieldFilter,
         return query;
     }
 
+    protected Map<String, Set<String>> merge(Map<String,Set<String>> a, Map<String,Set<String>> b) {
+        Map<String, Set<String>> map = new HashMap<>();
+        Set<String> keys = new HashSet<>();
+
+        keys.addAll(a.keySet());
+        keys.addAll(b.keySet());
+
+        for (String key: keys) {
+            Set<String> values = new HashSet<>();
+            if (a.get(key) != null) { values.addAll(a.get(key)); }
+            if (b.get(key) != null) { values.addAll(b.get(key)); }
+            map.put(key, values);
+        }
+
+        return map;
+    }
```

Comment:
@kevinschaper it would be nice to keep these "helper" functions off in another class and not in the Neo4jRepository... trying to keep separation of responsibilities. Maybe a MapUtils or QueryHelper, etc.